### PR TITLE
Create a function to recursively remove files

### PR DESCRIPTION
**Pull request by snwoods**

### CODE REVIEW

File: lib/xapi-stdext-unix/unixext.ml [hunk 1]
```diff
@@ -34,6 +34,27 @@ let mkdir_rec dir perm =
     mkdir_safe dir perm in
   p_mkdir dir
 
+(** removes a file or recursively removes files/directories below a directory without following
```
Comment (lindig) [inline lib/xapi-stdext-unix/unixext.ml:37]: We could add that providing a non-existent path will *not* lead to an error. So the behavior is similar to `rm -rf arg` in the shell.

File: lib/xapi-stdext-unix/unixext.ml [hunk 2]
```diff
@@ -34,6 +34,27 @@ let mkdir_rec dir perm =
...
+    match Unix.lstat path with
+    | exception Unix.Unix_error (Unix.ENOENT, _, _) ->
+        () (*noop*)
+    | exception e ->
+        raise e
+    | st -> (
+      match st.Unix.st_kind with
+      | Unix.S_DIR ->
+          Sys.readdir path |> Array.iter (fun file -> rm true (path // file)) ;
+          if rm_top then Unix.rmdir path
```
Comment (lindig) [inline lib/xapi-stdext-unix/unixext.ml:51]: I find it surprising that the `rm_top` case is handled in the recursive part and not at the top level where `rm` is called. I struggled to understand what it does in the recursive. I tested the code and it seems to do the right thing, though.

File: lib/xapi-stdext-unix/unixext.mli [hunk 3]
```diff
@@ -17,6 +17,7 @@ val _exit : int -> unit
 val unlink_safe : string -> unit
 val mkdir_safe : string -> Unix.file_perm -> unit
 val mkdir_rec : string -> Unix.file_perm -> unit
+val rm_rec : ?rm_top:bool -> string -> unit
```
Comment (lindig) [inline lib/xapi-stdext-unix/unixext.mli:20]: Documentation should go here.

**Review by lindig — APPROVED** [review]

File: lib/xapi-stdext-unix/unixext.ml (hunk 2 above)
Comment (snwoods) [reply lib/xapi-stdext-unix/unixext.ml:51]: Hmm I'm not sure how you could move rm_top outside the recursive part. The Unix.rmdir has to happen in the recursive part (and after its contents has been rm'd) and so rm_top is just a guard to stop the first iteration from doing Unix.rmdir if rm_top=false. Thereafter, rm_top=true in all the recursive calls to rm.

**Pull request merged**